### DISCUSSION TITLE
[EssentialsDiscord] How to add roles to people?

### DISCUSSION THREAD

**Discussion by HyperCodec** [EssentialsX/Essentials#5032] [Q&A]
I looked through the javadocs for EssentialsDiscord, I don't see any addRole method or anything similar, only getRoles and hasRole, neither of which have any support for role adding. Is there anything that actually adds roles or do I have to use a different API?

**Comment by pop4959** [comment]
EssentialsDiscord is not a general purpose Discord API. For that you would want to use something like [JDA](https://github.com/DV8FromTheWorld/JDA). However account linking in general is being worked on, and you can track the current progress of that in #4155.
